### DISCUSSION TITLE
Bark TTS incorrect credit

### DISCUSSION THREAD

**Discussion by gitmylo** [coqui-ai/TTS#2730] [General]
Hey, I was just browsing the TTS pypi page, and I was browsing the wiki after seeing that bark was added, and it seems credit for bark voice cloning [on this page](https://tts.readthedocs.io/en/dev/models/bark.html) is given to Serp-AI, even though I was the one who wrote the implementation? Serp-AI implemented it after I had made it, I was also specifically requested to, but the only official implementation is https://github.com/gitmylo/bark-voice-cloning-HuBERT-quantizer/.

Implementation source: https://github.com/gitmylo/bark-voice-cloning-HuBERT-quantizer/
Proof (From TTS itsself): https://github.com/coqui-ai/TTS/blob/21a3f280dea7ab7c6381f5805589514bea86da4d/TTS/tts/layers/bark/hubert/hubert_manager.py#L1
Serp-AI even has me listed as the creator of this cloning method under their repo: https://github.com/serp-ai/bark-with-voice-clone#shoutouts

I would really appreciate if this was cleared up. As I've lately been developing my first growing open source projects. (Mainly due to the before mentioned bark voice cloning). And I believe it would be helpful if the credits were adjusted to indicate the original project (which also provides instructions on how to implement voice cloning yourself, by the way).

**Comment by omkarade** [comment]
you can take the following steps:

Contact the maintainers of the TTS project and kindly explain the situation. Provide them with the relevant evidence, such as the implementation source and proof from the TTS code itself, as you mentioned in your message.

Request that they update the credits on the TTS PyPI page to accurately reflect the original implementation source and give proper credit to you for your contribution.

Be polite and respectful in your communication, emphasizing the importance of accurate credit and acknowledging the collaborative nature of open-source projects.

**Reply by gitmylo** [reply]
That was my attempt with this discussion, but i'm not sure if it will be read.

I wanted to create an issue, but I'm not entirely sure if it would fit a bug report. The other options would make even less sense. So I made a discussions post.

**Comment by gitmylo** [comment]
@erogol Can you please look at this? I believe that I'm the appropriate rights holder for the code and original implementation of Bark TTS HuBERT quantizer voice cloning. As the files even say. And I believe the original repository should be credited.

**Reply by erogol** [reply]
I agree. I'll fix this on Monday. 👍

**Reply by erogol** [reply]
@gitmylo one question. Have you tried fine-tuning Bark with your tokenizer?

**Reply by gitmylo** [reply]
I haven't, i believe SERP-AI did recently though.

**Reply by erogol** [reply]
If it is interesting for you, we can work together on that. So we don't just give you credits but have you onboard as one of the contributors 😄

**Reply by erogol** [reply]
@gitmylo does this look good for you? https://github.com/coqui-ai/TTS/pull/2733

**Reply by gitmylo** [reply]
> If it is interesting for you, we can work together on that. So we don't just give you credits but have you onboard as one of the contributors 😄

Sounds great.

> @gitmylo does this look good for you? #2733

yes, that looks good.

**Reply by erogol** [reply]
If you are on discord, let me know your user name and I get you there into the contributor's channel and we start working on fine-tuning.

**Reply by erogol** [reply]
BTW released the new version with the fix in the docs

**Reply by gitmylo** [reply]
> If you are on discord, let me know your user name and I get you there into the contributor's channel and we start working on fine-tuning.

My discord @ is @cheemsdogger, I don't think I'm currently in the discord server. But my friend requests are open!

**Reply by erogol** [reply]
@gitmylo sent you an invite
